File: WordSoulApi/WordSoulApi/Models/DTOs/Vocabulary/VocabularyDto.cs
using WordSoulApi.Models.Entities;

namespace WordSoulApi.Models.DTOs.Vocabulary
{
    public class VocabularyDto
    {
        public int Id { get; set; }
        public required string Word { get; set; }
        public required string Meaning { get; set; }
        public string? Pronunciation { get; set; } // e.g., "/wɜːrd/"
        public string? PartOfSpeech { get; set; }
        public string? CEFRLevel { get; set; } // Common European Framework of Reference for Languages level
        public string? Description { get; set; } // e.g., "A word used to describe something", "An action or state of being"
        public string? ExampleSentence { get; set; }
        public string? ImageUrl { get; set; }
        public string? PronunciationUrl { get; set; }

    }
}

File: wordsoul-mobile/src/types/VocabularyDto.ts
export interface VocabularyDto {
  id: number;
  word: string;
  meaning: string;
  imageUrl: string | null;
  pronunciation: string | null;
  pronunciationUrl: string | null;
  partOfSpeech: string;
  cefrLevel: string | null;
  description: string | null;
  exampleSentence: string | null;
  exampleSentenceAudioUrl: string | null;
}

export const VocabularyDifficultyLevelEnum = {
  Beginner: 0,
  Intermediate: 1,
  Advanced: 2,
} as const;

export type VocabularyDifficultyLevelEnum =
  (typeof VocabularyDifficultyLevelEnum)[keyof typeof VocabularyDifficultyLevelEnum];
